src/Geta.SEO.Sitemaps/XML/SitemapXmlGenerator.cs

```diff
@@ -564,5 +562,39 @@ protected bool IsAbsoluteUrl(string url, out Uri absoluteUri)
         {
             return Uri.TryCreate(url, UriKind.Absolute, out absoluteUri);
         }
+
+        protected bool TryGet<T>(ContentReference contentLink, out T content) where T : IContentData
+        {
+            content = default(T);
+            try
+            {
+                var status = this.ContentRepository.TryGet<T>(contentLink, out var local);
+                content = (T)local;
+                return status;
+            }
+            catch (Exception e)
+            {
+                Log.Error("Error on contentReference " + contentLink.ID + Environment.NewLine + e);
+            }
+
+            return false;
+        }
+
+        protected bool TryGet<T>(ContentReference contentLink, LoaderOptions settings, out T content) where T : IContentData
```

Comment:
Could you merge these two methods? And make LoaderOptions an optional parameter?

Reply:
I think I manged to reduce the code